snuba/datasets/__init__.py

```diff
@@ -1,12 +1,33 @@
-from typing import Optional, Mapping, Sequence, Tuple
+from typing import (
+    Any,
+    Optional,
+    Mapping,
+    Sequence,
+    Tuple,
+)
+from typing_extensions import Protocol
 
 from snuba.datasets.dataset_schemas import DatasetSchemas
 from snuba.datasets.table_storage import TableWriter
 from snuba.query.extensions import QueryExtension
+from snuba.request import Request
 
 from snuba.util import escape_col, parse_datetime
 
 
+class SplitQueryFunc(Protocol):
```

Comment:
Was having a hard time defining the function definition for `parse_and_run_query` and then found `Protocol`: https://mypy.readthedocs.io/en/latest/protocols.html

Reply:
Curious to know if you feel there is a better spot for `SplitQueryFunc`.

Reply:
Is this equivalent to:
```
Callable[[Dataset, Request, Any, Any], Tuple[Mapping[str, Any], int]]
```

Plus something to define self